diff --git a/dist/MarkdownEditor.svelte b/dist/MarkdownEditor.svelte
index 679fad89f0022c0a3c90c0a463b46d1aaa935f82..e613c1ffe16e74b8578c3adca681c6b53431406a 100644
--- a/dist/MarkdownEditor.svelte
+++ b/dist/MarkdownEditor.svelte
@@ -13,6 +13,7 @@ export let disableToolbar = false;
 export let placeholder = "";
 export let textarea = {};
 export let selectedTab = "write";
+export let onKeydown;
 let userLabels = {};
 export { userLabels as labels };
 const labels = {
@@ -93,6 +94,7 @@ onMount(() => carta.$setElement(editorElem));
 					bind:value
 					bind:resize={resizeInput}
 					bind:elem={inputElem}
+					{onKeydown}
 				>
 					<!-- Input extensions components -->
 					{#if mounted}
diff --git a/dist/MarkdownEditor.svelte.d.ts b/dist/MarkdownEditor.svelte.d.ts
index 99f03fb3c1d45e9d5e4811854fa5e8ff3d7ae74a..6956735d253eb08e728155b4a466362299a95538 100644
--- a/dist/MarkdownEditor.svelte.d.ts
+++ b/dist/MarkdownEditor.svelte.d.ts
@@ -14,6 +14,7 @@ declare const __propDef: {
         textarea?: TextAreaProps | undefined;
         selectedTab?: "preview" | "write" | undefined;
         labels?: Partial<Labels> | undefined;
+        onKeydown?: ((e: KeyboardEvent & { currentTarget: EventTarget & HTMLTextAreaElement }) => any) | undefined;
     };
     events: {
         [evt: string]: CustomEvent<any>;
diff --git a/dist/internal/components/Input.svelte b/dist/internal/components/Input.svelte
index caccbe8e0e019ada654dc725bf683b8aa37b6ad4..ea9fa6f1909921639da0debf5209347c00cc4a1c 100644
--- a/dist/internal/components/Input.svelte
+++ b/dist/internal/components/Input.svelte
@@ -7,6 +7,7 @@ export let placeholder = "";
 export let elem;
 export let handleScroll;
 export let props = {};
+export let onKeydown;
 let textarea;
 let highlighElem;
 let highlighted = value;
@@ -101,6 +102,7 @@ onMount(setInput);
 			bind:value
 			bind:this={textarea}
 			on:scroll={() => (textarea.scrollTop = 0)}
+			on:keydown={onKeydown}
 		/>
 	</div>
 
diff --git a/dist/internal/components/Toolbar.svelte b/dist/internal/components/Toolbar.svelte
index 075440b134c31e395b6565cdd237c05285f9b427..9111b3b480a88ffb1cafcf5003b4743064ff7049 100644
--- a/dist/internal/components/Toolbar.svelte
+++ b/dist/internal/components/Toolbar.svelte
@@ -18,6 +18,7 @@ let showMenu = false;
 const IconPadding = 8;
 const waitForDOMUpdate = () => new Promise(requestAnimationFrame);
 const onResize = debounce(async () => {
+if (!toolbar || !iconsContainer) return;
   const overflowing = () => toolbar.scrollWidth - toolbar.clientWidth > 0;
   while (overflowing()) {
     visibleIcons.pop();
