diff --git a/dist/index.mjs b/dist/index.mjs
index 767438879c2e425ebad65f53453ae755e105174b..8edd58b05d8fd30faa9b7f68fa0dbb80ff74c519 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -21,6 +21,8 @@ function setup(markdownit, highlighter, options) {
     highlightLines = true
   } = options;
   markdownit.options.highlight = (code, lang = "text", attrs) => {
+    if (!highlighter.getLoadedLanguages().includes(lang)) lang = "text";
+
     const codeOptions = {
       ...options,
       lang
diff --git a/package.json b/package.json
index a0d1a4d3ca3392efec51b3e9853ed12016bb4a3d..638673b4746e17348b751ee991fef4156bc2e3ba 100644
--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "dist"
   ],
   "dependencies": {
-    "markdown-it": "^13.0.2",
+    "markdown-it": "^14.0.0",
     "shikiji": "0.8.0"
   },
   "devDependencies": {
